test/integration/scheduler/predicates_test.go

```diff
@@ -870,3 +870,68 @@ func TestInterPodAffinity(t *testing.T) {
 		}
 	}
 }
+
+// TestNodePIDPressue verifies that scheduler's CheckNodePIDPressurePredicate predicate
+// functions works correctly.
+func TestNodePIDPressue(t *testing.T) {
+	context := initTest(t, "node-pid-pressure")
+	defer cleanupTest(t, context)
+	// Add a few nodes.
+	nodes, err := createNodes(context.clientSet, "testnode", nil, 2)
+	if err != nil {
+		t.Fatalf("Cannot create nodes: %v", err)
+	}
+
+	cs := context.clientSet
+
+	// Adds PID pressure condition to the frist node.
+	nodes[0].Status.Conditions = []v1.NodeCondition{
+		{
+			Type:   v1.NodePIDPressure,
+			Status: v1.ConditionTrue,
+		},
+	}
+
+	// So the second node is the target node.
+	targetNode := nodes[1].Name
+
+	// Creats test pod.
+	testPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: "fake-name"},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{Name: "container", Image: imageutils.GetPauseImageName()},
+			},
+		},
+	}
+
+	testPod, err = cs.CoreV1().Pods(context.ns.Name).Create(testPod)
+	if err != nil {
+		t.Fatalf("Test Failed: error: %v, while creating pod", err)
+	}
+
+	err = wait.Poll(pollInterval, wait.ForeverTestTimeout, podScheduled(cs, testPod.Namespace, testPod.Name))
+	if err != nil {
+		t.Errorf("Test Failed: error, %v, while waiting for scheduled", err)
+	}
+
+	// Retrieves the pod and check its target node.
+	pod, err := cs.CoreV1().Pods(testPod.Namespace).Get(testPod.Name, metav1.GetOptions{})
+	if errors.IsNotFound(err) || err != nil {
+		t.Errorf("Test Faield: error, %v, while retrieving pod", err)
+	}
+
+	if pod.Spec.NodeName != targetNode {
+		t.Errorf("Test Failed: pod %v/%v is not scheduled to the target node %s",
+			pod.Namespace, pod.Name, targetNode)
+	}
+
+	err = cs.CoreV1().Pods(context.ns.Name).Delete(testPod.Name, metav1.NewDeleteOptions(0))
```

Comment:
Deleting the pods is not necessary in this test. You can leave it to the `cleanupTest` to take care of pod cleanup.

Reply:
The `cleanupTest` only cleanup nodes :)